### PR TITLE
Set up salesforce tracking lambda

### PR DESCRIPTION
**Pull request by jonflynng**
Create a lambda and SQS queue to handle the tracking of product-switches in Salesforce. The lambda is triggered by a new message on the queue.

### CODE REVIEW

File: handlers/product-move-api/src/main/scala/com/gu/productmove/salesforce/Salesforce.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package com.gu.productmove.salesforce
```
Comment (jonflynng) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/salesforce/Salesforce.scala:1]: None of this is new, just moved to new file

File: handlers/product-move-api/src/main/scala/com/gu/productmove/salesforce/SalesforceHandler.scala [hunk 2]
```diff
@@ -5,46 +5,32 @@ import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
...
-      previousAmount: BigDecimal,
-      previousRatePlanName: String,
-      newRatePlanName: String,
-      requestedDate: LocalDate,
-      effectiveDate: LocalDate,
-      refundAmount: BigDecimal,
-  )
+class SalesforceHandler extends RequestHandler[SQSEvent, Unit] {
+  override def handleRequest(input: SQSEvent, context: Context): Unit = {
+    val records: List[SQSEvent.SQSMessage] = input.getRecords.asScala.toList
```
[the remark this answers is not in the document]
Comment (rupertbates) [reply handlers/product-move-api/src/main/scala/com/gu/productmove/salesforce/SalesforceHandler.scala:19]: They are batched to the `BatchSize` specified in the cloudformation for `SQSTriggerSF`

File: handlers/product-move-api/riff-raff.yaml [hunk 3]
```diff
@@ -23,4 +23,5 @@ deployments:
       functionNames:
       - move-product-
```
Comment (rupertbates) [inline handlers/product-move-api/riff-raff.yaml:24]: I think we should rename this lambda to match the others, eg. product-switch-something doesn't have to be this pr though

File: handlers/product-move-api/src/main/scala/com/gu/productmove/SQS.scala [hunk 4]
```diff
@@ -14,6 +15,8 @@ trait SQS {
   def sendEmail(message: EmailMessage): ZIO[Any, String, Unit]
```
Comment (rupertbates) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/SQS.scala:15]:
```suggestion
  def sendEmail(message: EmailMessage): IO[String, Unit]
```

File: handlers/product-move-api/src/main/scala/com/gu/productmove/SQS.scala [hunk 5]
```diff
@@ -14,6 +15,8 @@ trait SQS {
...
 
   def queueRefund(refundInput: RefundInput): ZIO[Any, String, Unit]
```
Comment (rupertbates) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/SQS.scala:17]:
```suggestion
  def queueRefund(refundInput: RefundInput): IO[String, Unit]
```

File: handlers/product-move-api/src/main/scala/com/gu/productmove/SQS.scala [hunk 6]
```diff
@@ -14,6 +15,8 @@ trait SQS {
...
+
+  def queueSalesforceTracking(salesforceRecordInput: SalesforceRecordInput): ZIO[Any, String, Unit]
```
Comment (rupertbates) [inline handlers/product-move-api/src/main/scala/com/gu/productmove/SQS.scala:19]:
```suggestion
  def queueSalesforceTracking(salesforceRecordInput: SalesforceRecordInput): IO[String, Unit]
```

**Review by rupertbates — APPROVED** [review]
Couple of suggestions for simpler types but looks good 👍

**Comment by jonflynng** [comment]
[Implement product-switch tracking](https://trello.com/c/rM85Wz4m/449-implement-product-switch-tracking)

**Pull request merged**